### ISSUE TITLE
Keyboard input not working in iOS 7

### ISSUE THREAD

**Issue by drlippman** [mathquill/mathquill#559]
OK, I know iOS 7.0.3 is way old, so maybe it's not worth worrying about, but for whatever reason the on-screen keyboard doesn't seem to be able to input into MathQuill.  I tried it on the mathquill.com page input box, and typing produced no characters, both in Safari and Chrome.

I checked, and keydown and keypress events are being fired in textareas, so not sure why it would be having problems.

**Comment by drlippman** [comment]
I'm having issues with other functions too, like API calls, so maybe there's a larger incompatibility. Unfortunately I don't have the necessary setup to debug properly.

**Comment by drlippman** [comment]
Interesting. I'm having similar problems using the onscreen keyboard to type in the box at mathquill.com using my Android Nexus 5x running Chrome.

**Comment by laughinghan** [comment]
I mean it's worrying to me...

**Comment by laughinghan** [comment]
which version of Android Chrome? Think you could get the User Agent from something like http://whatsmyuseragent.com/ ?

**Comment by drlippman** [comment]
Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29Q) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.95 Mobile Safari/537.36
On Mar 9, 2016 8:58 PM, Han Seoul-Oh <notifications@github.com> wrote:which version of Android Chrome? Think you could get the User Agent from something like http://whatsmyuseragent.com/ ?

**Comment by drlippman** [comment]
I updated my iPad to iOS 9 and typing using the onscreen keyboard is working fine now, so it must have been an issue in iOS 7.  Still having the issue on my Andriod.

**Comment by The-Penultimate-Defenestrator** [comment]
Text selection still does not work on iOS 9. (used for things like selecting text and pressing `/` for putting things in the numerator)

**Comment by laughinghan** [comment]
Text selection like the magnifying loupe and the two cursors at either end of the selection with the draggable dots? We have no plans to implement that. That was introduced relatively recently compared to the other system features we emulate (2009, in iOS 3), isn't standardized across mobile platforms (Android has handles rather than dots, no magnifying loupe, and a cut/copy/paste icon bar rather than a pop-up menu; Windows Phone has little arrows rather than dots, also no magnifying loupe, and a pop-up copy icon), and may well change in the future (the "Cut | Copy | Paste" pop-up menu has changed at least once, from the glossy-curved look to the flat design look; I don't think they've changed the look of the selection itself but I wouldn't put it past Apple).

Options include:
- browser sniff and emulate each platform's text selection interface (lots of work, unclear future-proof-ness)
- emulate just one platform's text selection interface on all platforms (possibly tractable, but it'll look silly at best on the other platforms, and also has future-proofing problems)
- implement an interface with the lowest-common-denominator features (probably drag handles, no loupe, and cut/copy/paste pop-up menu; probably actually tractable and will be at least usable well into the future, even if it's not that pretty)
- do nothing since the poor usability of mobile keyboards with MathQuill means this probably wouldn't make a difference.

Given how hard it is to type non-letters&mdash;hell, even to type non-words!&mdash;on mobile keyboards, we're currently taking the last approach.

**Comment by The-Penultimate-Defenestrator** [comment]
I’d love to see more mobile optimization. Autocorrect (at least on iOS
Safari) can be disabled for text boxes., which makes it easier to type
non-words. Non-letters (I assume you’re referring to putting in backslash
codes) are much easier if the interface implements buttons as shortucts for
putting in \sqrt, \pi, etc.

**Comment by laughinghan** [comment]
Good point, autocorrect is supposed to already be disabled when typing in MathQuill. Desmos has a custom keyboard for iOS with button shortcuts for \sqrt, \pi, etc&mdash;have you seen it? What do you think of that?

**Comment by The-Penultimate-Defenestrator** [comment]
I've got Desmos on my iPad, not my iPhone, I'm downloading it on my phone now to take a look.

**Comment by laughinghan** [comment]
The Desmos custom keyboard is on the web, you should see the same thing if you go to desmos.com/calculator on an iPad or iPhone.

Indeed, some form of usable text selection isn't intractable, and it'd be delightful if someone did it, but we have no plans to.

**Comment by The-Penultimate-Defenestrator** [comment]
Any idea how Desmos implemented this on web? If not, I'll try to figure it out myself, I just wanted to see if you knew first.

**Comment by laughinghan** [comment]
I do, I helped them with it. Let's move this to another thread though

**Comment by nscozzaro** [comment]
I'm having issues with the Mathquill box on Android (pixel 2). I select the box and type on my phone's keyboard but the box remains blank. Please advise on how this issue may be addressed.
